app/mailers/user_mailer.rb

```diff
@@ -89,6 +89,21 @@ def work_package_watcher_added(work_package, user, watcher_setter)
     end
   end
 
+  def work_package_watcher_removed(work_package, user, watcher_remover)
+    User.execute_as user do
+      @issue = work_package
+      @watcher_remover = watcher_remover
+
+      set_work_package_headers(work_package)
+      message_id work_package, user
+      references work_package, user
+
+      with_locale_for(user) do
+        mail to: user.mail, subject: subject_for_work_package(work_package)
+      end
+    end
+  end
+
```

Comment:
The only difference compared to [`work_pacakge_patcher_added`](https://github.com/opf/openproject/pull/7927/files#diff-392461f2d7c20a4d5e9f5409e0125135R77) is the different naming of the instance variable (`@watcher_remover` instead of `@watcher_setter`). I believe that both methods can be unified if the variable is renamed e.g. to `@acting_user`. In other places `watcher_changer` has already been used so that might also be an appropriate name.

Once the variable and by that the method is unified, the templates can be unified as well via partials as the only difference between the adding and removing notification mail is the i18n key.